ols/app/main.py

```diff
@@ -41,12 +41,13 @@
 @app.middleware("")
 async def rest_api_counter(request: Request, call_next):
     """Middleware with REST API counter update logic."""
+    response = await call_next(request)
     # ignore /metrics endpoint that will be called periodically
     path = request.url.path
     if not path.endswith("/metrics/"):
         # just update metrics
-        metrics.rest_api_calls.inc()
-    return await call_next(request)
+        metrics.rest_api_calls_total.labels(response.status_code).inc()
```

Comment:
seems like we should included the path in the labels, so we can differentiate the types of api calls being made.

Reply:
yup that would be super easy. I'll create PR later today

Reply:
A reminder that we should put the path leading to the endpoint function without query parameters, for example: 
we label with  `/version/endpoint`  instead of `/version/endpoint?arg1=val1...` or `/version/endpoint/arg1/val1`
A distinct value in the path label is an additional cardinality to the metric, making the Prometheus consumes more storage and computation.

Reply:
@bparees https://github.com/openshift/lightspeed-service/pull/432

Reply:
> A reminder that we should put the path leading to the endpoint function without query parameters, for example: we label with `/version/endpoint` instead of `/version/endpoint?arg1=val1...` or `/version/endpoint/arg1/val1` A distinct value in the path label is an additional cardinality to the metric, making the Prometheus consumes more storage and computation.

path does not contain query params

Reply:
great, that simplifies our work :)